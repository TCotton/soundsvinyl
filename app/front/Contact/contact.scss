@import '../../assets/sass/tools';

.contactPage {
	text-align: center;
	margin: 50px 0 75px 0;

	h2 {
		@include font-calculator($font_family_header, 45px, 0.5);
		color: $secondHeaderColour;
		line-height: 1.2em;
	}

	p {
		@include font-calculator($font_family_body, 16px);
		color: $paragraphColour;
		line-height: 1.3em;
	}

	form {
		max-width: 480px;
		margin: 0 auto;
	}

	label {
		display: inline-block;
		@include font-calculator($font_family_body, 18px, 0.25);
		font-weight: 300;
	}

	input {
		display: inline-block;
		width: 100%;
		border: 1px solid $contactInputBorder;
		min-height: 50px;
		padding: 10px;
	}

	textarea {
		display: inline-block;
		width: 100%;
		border: 1px solid $contactInputBorder;
		min-height: 200px;
		padding: 15px;
	}

	input,
	textarea {
		@include font-calculator($font_family_body, 16px, 0);
	}

	input:not([type='submit']),
	textarea {
		margin-bottom: 30px;
	}

	input[type='submit'] {
		@include font-calculator($font_family_body, 18px, 0);
		color: white;
		background: $commentsSubmitBackground;
	}
}
