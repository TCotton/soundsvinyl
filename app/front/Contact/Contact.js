import React, { Component } from 'react';
import './contact.scss';
import axios from 'axios';
import { homeURI } from '../../helper_constants';

class Contact extends Component {

	constructor ( props ) {
		super( props );

		this.state = {
			contactName: '',
			contactEmail: '',
			contactMessage: '',
			error: null,
			message: null,
			zipcode: ''
		};

		this.handleSubmit = this.handleSubmit.bind( this );
		this.handleInputChange = this.handleInputChange.bind( this );
	}

	handleSubmit ( e ) {
		e.preventDefault();

		const {
			contactEmail,
			contactName,
			contactMessage,
		} = this.state;

		axios.post( `${homeURI}/apiV1/sendmail`, { contactEmail, contactName, contactMessage } )
			.then( res => {

				if( res.data.error ) {
					this.setState( { error: res.data.error } );
				}

				if( res ) {
					if( res.data.success ) {
						this.setState( { message: res.data.message } , () => {
							document.querySelector('body').scrollTop = 0;
						});
					}
				}

			}).catch( ( e ) => {
			this.setState( { error: e.toString() } );
		})
	}

	handleInputChange (event) {

		const target = event.target;
		const value = target.value;
		const name = target.name;

		this.setState({
			[name]: value
		});

	}

	render () {

		const {
			message,
			error,
			contactEmail,
			contactName,
			contactMessage,
			zipcode,
		} = this.state;

		return (
			<main styleName='contactPage'>
				<h2>
					Contact us
				</h2>
				<p>
					Please contact us if you any recommendations for records to review
				</p>

				{error &&
					<p className='error'>
						{error}
					</p>}

				{message &&
				<p className='message'>
					{message}
				</p>}

				<form onSubmit={this.handleSubmit}>
					<label htmlFor='contactName'>
						Your name
					</label>
					<input
						id='contactName'
						maxLength='254'
						name='contactName'
						onChange={this.handleInputChange}
						required
						type='text'
						value={contactName}
					/>

					<label htmlFor='contactEmail'>
						Your email
					</label>
					<input
						id='contactEmail'
						maxLength='254'
						name='contactEmail'
						onChange={this.handleInputChange}
						required
						type='email'
						value={contactEmail}
					/>

					<label htmlFor='contactMessage'>
						Your message
					</label>
					<textarea
						id='contactMessage'
						maxLength='500'
						minLength='10'
						name='contactMessage'
						onChange={this.handleInputChange}
						required
						value={contactMessage}
					/>

					<input
						disabled={message}
						name='contactSubmit'
						type='submit'
						value='submit'
					/>

					<span
						aria-hidden='true'
						className='hide'
					>
						<label
							aria-hidden='true'
							className='hide'
							htmlFor='zipcode'
						>
							Your zipcode
						</label>
						<input
							aria-hidden='true'
							className='hide'
							id='zipcode'
							name='zipcode'
							onChange={this.handleInputChange}
							type='text'
							value={zipcode}
						/>
					</span>

				</form>
			</main>
		)
	}
}

export default Contact;
