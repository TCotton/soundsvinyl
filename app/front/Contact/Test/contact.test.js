import React from 'react';
import Contact from '../Contact';
import renderer from 'react-test-renderer';

describe( 'Component', () => {
	let component;
	beforeEach( () => {
		component = renderer.create(
			<Contact />
		);
	});

	describe( 'Contact', () => {
		it( 'should be defined', () => {
			expect( component ).toBeDefined();
		});
	});
});
