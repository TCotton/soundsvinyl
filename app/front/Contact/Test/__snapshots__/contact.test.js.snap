// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Contact should render correctly 1`] = `
<main
  className="contact__contactPage___21tug"
>
  <h2>
    Contact us
  </h2>
  <p>
    Please contact us if you any recommendations for records to review
  </p>
  <form
    onSubmit={[Function]}
  >
    <label
      htmlFor="contactName"
    >
      Your name
    </label>
    <input
      id="contactName"
      maxLength="254"
      name="contactName"
      onChange={[Function]}
      required={true}
      type="text"
      value=""
    />
    <label
      htmlFor="contactEmail"
    >
      Your email
    </label>
    <input
      id="contactEmail"
      maxLength="254"
      name="contactEmail"
      onChange={[Function]}
      required={true}
      type="email"
      value=""
    />
    <label
      htmlFor="contactMessage"
    >
      Your message
    </label>
    <textarea
      id="contactMessage"
      maxLength="500"
      minLength="10"
      name="contactMessage"
      onChange={[Function]}
      required={true}
      value=""
    />
    <input
      data-test-dataid="contactSubmit"
      disabled={null}
      name="contactSubmit"
      type="submit"
      value="submit"
    />
    <span
      aria-hidden="true"
      className="hide"
    >
      <label
        aria-hidden="true"
        className="hide"
        htmlFor="zipcode"
      >
        Your zipcode
      </label>
      <input
        aria-hidden="true"
        className="hide"
        id="zipcode"
        name="zipcode"
        onChange={[Function]}
        type="text"
        value=""
      />
    </span>
  </form>
</main>
`;
